src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1179,6 +1179,15 @@ public StringBuilder Append(char value)
                 return this;
             }
 
+            unsafe
+            {
+                Span<char> tempBuffer = stackalloc char[32]; // should be enough for all primitives
+                if (value.TryFormat(tempBuffer, out int charsWrittenToTmp, format: default, provider: null))
```

Comment:
Have you measured performance here?  In the common case we should be formatting directly into the StringBuilder's buffer and only falling back to allocating in the rarer case where there's no enough space in the buffer.  With this change, we're now formatting into a temporary buffer and then copying from there into the builder; that'll help with the rarer case but could hurt the common case.

Reply:
Oh, nevermind, GitHub was hiding the previous code; I thought you'd replaced it, but you instead just added this case.